### DISCUSSION TITLE
trying to fix my local docker setup

### DISCUSSION THREAD

**Discussion by willenglishiv** [yuezk/GlobalProtect-openconnect#412] [Q&A]
Hi everyone,

So I recently followed these instructions: 
https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/#Linux

to get my local kubernetes to respond to real URLs without having to add them all manually to my hosts file.

The method works with VPN turned off, but I need it to work with it turned on.

I know the VPNC script mods my resolved.conf file and removes my local nameserver, I'm using the command line tool.  Is there any way to make this work? I'm also on Ubuntu 24, previously I was directed to use dnsmasq on my linux pc but I ran into similar issues.

**Comment by yuezk** [comment]
@willenglishiv do you mean that the VPNC overwrites your DNS configurations so your customizations no longer work?

**Comment by willenglishiv** [comment]
yes it does.

**Reply by willenglishiv** [reply]
my current googlefu is pointing me to using vpn-slice or something to that effect, but I was wondering if you had a way to fix this without having to write a script.  If I really wanted to be annoying I could just add my nameserver on top of what the VPN links look for but that would have to be every time. (Also, this is isolated to a Virtual Machine btw)

**Reply by yuezk** [reply]
I saw lots of people use [vpn-slice](https://github.com/dlenski/vpn-slice) with this client. But I don't have this scenario, so I'm not familiar with how to resolve this.
